### PR TITLE
Install nightlies from Azure

### LINKED ISSUE #10993: DOC: astropy nightly wheels
Dev doc for nightly wheels. How does one use it? How does one maintain it?

Follow up of #10405

cc @astrofrog and @bsipocz

### PR DESCRIPTION
**Pull request by Cadair**
I have no idea if this is the place in the docs where this should be, but hey.

EDIT: Fix #10993

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
    <https://github.com/astropy/astropy/issues?q=+label%3ACASA-Installation+>`_
    and if you do not encounter your issue there, please post a new one.
 
+
+Installing pre-built Development Versions of ``astropy``
+--------------------------------------------------------
+
+Most nights a development snapshot of ``astropy`` will be compiled.
```
Comment (pllim) [inline docs/install.rst:400]: Do you think it is useful to provide a link for downstream developers to check the build logs? For example, personally, I would like to see when did the last "nightly" happen.

Also, is this available for all OSes? Or just Linux?

File: docs/install.rst [hunk 2]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
...
+This is useful if you want to test against a development version of astropy but
+do not want to have to build it yourself.
+
+Installing these "nighlies" of ``astropy`` can be achieved by using pip::
```
Comment (pllim) [inline docs/install.rst:404]:
```suggestion
Installing these "nightlies" of ``astropy`` can be achieved by using ``pip``::
```

File: docs/install.rst [hunk 3]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
...
+
+  $ pip install --extra-index-url=https://pkgs.dev.azure.com/astropy-project/astropy/_packaging/nightly/pypi/simple/ --pre astropy
+
+The extra index URL tells pip to check the pip index on Azure where the
```
Comment (pllim) [inline docs/install.rst:408]:
```suggestion
The extra index URL tells ``pip`` to check the ``pip`` index on Azure Pipelines, where the
```

File: docs/install.rst [hunk 4]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
...
+nightlies are built, and the ``--pre`` command tells pip to install pre-release
```
Comment (pllim) [inline docs/install.rst:409]:
```suggestion
nightlies are built, and the ``--pre`` command tells ``pip`` to install pre-release
```

File: docs/install.rst [hunk 5]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
...
+versions (in this case ``.dev`` releases).
```
Comment (pllim) [inline docs/install.rst:410]: So, when we push out RC, will `--pre` favor the RC over dev, or not?

File: docs/install.rst (hunk 1 above)
Comment (Cadair) [reply docs/install.rst:400]: All the binaries that are built for a release are built. You can see the list of available versions (and their upload date) here: https://dev.azure.com/astropy-project/astropy/_packaging?_a=package&feed=nightly&package=astropy&protocolType=PyPI&version=4.3.dev694&view=versions

File: docs/install.rst (hunk 5 above)
Comment (Cadair) [reply docs/install.rst:410]: yes, those version numbers are higher.

File: docs/install.rst (hunk 1 above)
Comment (pllim) [reply docs/install.rst:400]: Should we add that link to the doc?

File: docs/install.rst (hunk 5 above)
Comment (pllim) [reply docs/install.rst:410]: But you don't push RC to Azure, right? So maybe there's no ambiguity there?

**Comment by pllim** [comment]
> I have no idea if this is the place in the docs where this should be

I think this is as good as any. Though after this is merged, should update `package-template` too.

File: docs/install.rst [hunk 6]
```diff
@@ -393,6 +393,22 @@ packages that may be useful for radio astronomy).
    <https://github.com/astropy/astropy/issues?q=+label%3ACASA-Installation+>`_
    and if you do not encounter your issue there, please post a new one.
 
+
+Installing pre-built Development Versions of ``astropy``
+--------------------------------------------------------
+
+Most nights a development snapshot of ``astropy`` will be compiled.
+This is useful if you want to test against a development version of astropy but
+do not want to have to build it yourself.
```
Comment (pllim) [inline docs/install.rst:402]:
```suggestion
do not want to have to build it yourself. You can see the
`available astropy dev snapshots page <https://dev.azure.com/astropy-project/astropy/_packaging?_a=package&feed=nightly&package=astropy&protocolType=PyPI&view=versions>`_
to find out what is currently being offered.
```

File: docs/install.rst [hunk 7]
```diff
@@ -393,6 +393,24 @@ packages that may be useful for radio astronomy).
    <https://github.com/astropy/astropy/issues?q=+label%3ACASA-Installation+>`_
    and if you do not encounter your issue there, please post a new one.
 
+
+Installing pre-built Development Versions of ``astropy``
+--------------------------------------------------------
+
+Most nights a development snapshot of ``astropy`` will be compiled.
+This is useful if you want to test against a development version of astropy but
+do not want to have to build it yourself. You can see the
```
Comment (pllim) [inline docs/install.rst:402]:
```suggestion
do not want to have to build it yourselves. You can see the
```

**Review by pllim — APPROVED** [review]

**Pull request merged**